packages/discovery-provider/src/queries/get_follow_intersection_users.py

```diff
@@ -1,51 +1,41 @@
-from src.models.social.follow import Follow
-from src.models.users.user import User
-from src.queries import response_name_constants
-from src.queries.query_helpers import (
-    get_current_user_id,
-    paginate_query,
-    populate_user_metadata,
-)
-from src.utils import helpers
+from sqlalchemy.sql import text
+
+from src.queries.get_unpopulated_users import get_unpopulated_users
+from src.queries.query_helpers import populate_user_metadata
 from src.utils.db_session import get_db_read_replica
 
+sql = text(
+    """
+select
+  x.follower_user_id
+from follows x
+join aggregate_user au on x.follower_user_id = au.user_id
+join follows me
+  on me.follower_user_id = :my_id
+  and me.followee_user_id = x.follower_user_id
+  and me.is_delete = false
+where x.followee_user_id = :other_user_id
+  and x.is_delete = false
+order by follower_count desc
+limit :limit
+offset :offset
+"""
+)
 
-def get_follow_intersection_users(args):
-    followee_user_id = args.get("followee_user_id")
-    follower_user_id = args.get("follower_user_id")
 
+def get_follow_intersection_users(args):
     users = []
+    my_id = args.get("my_id")
+
     db = get_db_read_replica()
     with db.scoped_session() as session:
-        query = session.query(User).filter(
-            User.is_current == True,
-            User.user_id.in_(
-                session.query(Follow.follower_user_id)
-                .filter(
-                    Follow.followee_user_id == followee_user_id,
-                    Follow.is_current == True,
-                    Follow.is_delete == False,
-                )
-                .intersect(
-                    session.query(Follow.followee_user_id).filter(
-                        Follow.follower_user_id == follower_user_id,
-                        Follow.is_current == True,
-                        Follow.is_delete == False,
-                    )
-                )
-            ),
-        )
-        users = paginate_query(query).all()
-        users = helpers.query_result_to_list(users)
-        user_ids = [user[response_name_constants.user_id] for user in users]
-        current_user_id = get_current_user_id(required=False)
+        rows = session.execute(sql, args)
```

Comment:
I am in favor of this pattern of using sql text, but it would be nice if we could (with a small amount of effort):
* Define the expected arguments in a typed structure of some sort
* Maybe explicitly read the arguments above and construct the input to this manually so it's clear what values are being sent into the SQL (and also so we are only sending values that are used in the parameters for the sql above)

As it is, it's hard to tell when calling this query function what values need to be provided and feels easy to mess that up at the API layer.